prime-router/src/main/kotlin/azure/WorkflowEngine.kt

```diff
@@ -158,6 +160,10 @@ class WorkflowEngine(
         return blob.downloadBlob(header.task.bodyUrl)
     }
 
+    fun recordAction(actionHistory: ActionHistory) {
+        actionHistory.saveToDb(db)
+    }
+
```

Comment:
By passing this through WorkflowEngine, it keeps the database access stuff hidden from the Function level.

NOTE:  Right now, the Action History stuff is being committed in a separate transaction from the original data, blob creation etc.   That needs to be cleaned up.